### ISSUE TITLE
[Bug]: setuptools no longer bundled with Python

### ISSUE THREAD

**Issue by twonfi** [ThioJoe/YT-Spammer-Purge#1088]
### Duplicate Issues

- [X] There are **no existing posts relating to my problem**
- [ ] There are **existing posts relating to my problem, but the solution given, doesn't work for me.**

### What happened?

As of Python 3.12, setuptools (`pkg_resources`) was no longer bundled (see python/cpython#95299) and needs to be manually installed using pip.

The Python script tries to import `pkg_resources` (included in setuptools). However, it failed as setuptools is no longer bundled with Python 3.12. After installing setuptools, the script works as usual.

### Release version

v2.17.0

### Steps to reproduce

1. Install requirements `pip3 install -r requirements.txt`
2. Run the script `python3 YTSpammerPurge.py`
3. Shows error that `pkg_resources` was not found
4. Install setuptools `pip3 install setuptools`
5. Rerun `python3 YTSpammerPurge.py`
6. Works; shows login screen

### What platform are you seeing this problem on?

MacOS

**Comment by PacoBell** [comment]
Yup, I ran into this bug even on Python 3.11. Install script should really check for this and automatically install.

**Comment by twonfi** [comment]
Appears to be fixed by ac2ac53557237011c1d2bed3c2e9455a8f7e5f4e.
